core/arch/arm/kernel/boot.c

```diff
@@ -1296,13 +1308,19 @@ static void init_primary(unsigned long pageable_part, unsigned long nsec_entry)
  * Note: this function is weak just to make it possible to exclude it from
  * the unpaged area.
  */
-void __weak boot_init_primary_late(unsigned long fdt)
+void __weak boot_init_primary_late(unsigned long __maybe_unused fdt)
 {
+#ifdef CFG_EXTERNAL_DT
```

Comment:
I think we should remove this condition and change `init_external_dt()`/`update_external_dt()` to do nothing when `!IS_ENABLED(CFG_EXTERNAL_DT)`.